cmd/snap/cmd_set_test.go

```diff
@@ -109,6 +109,35 @@ func (s *snapSetSuite) TestSnapSetIntegrationStringWithExclamationMark(c *check.
 	c.Check(s.setConfApiCalls, check.Equals, 1)
 }
 
+func (s *snapSetSuite) TestSnapSetParseStrictJSON(c *check.C) {
+	// mock server
+	s.mockSetConfigServer(c, map[string]interface{}{"a": "b", "c": json.Number("1"), "d": map[string]interface{}{"e": "f"}})
+
+	_, err := snapset.Parser(snapset.Client()).ParseArgs([]string{"set", "snapname", "-t", `key={"a":"b", "c":1, "d": {"e": "f"}}`})
+	c.Assert(err, check.IsNil)
+	c.Check(s.setConfApiCalls, check.Equals, 1)
+}
+
+func (s *snapSetSuite) TestSnapSetFailParsingWithStrictJSON(c *check.C) {
+	_, err := snapset.Parser(snapset.Client()).ParseArgs([]string{"set", "snapname", "-t", `key=notJSON`})
+	c.Assert(err, check.ErrorMatches, "failed to parse JSON:.*")
+}
+
+func (s *snapSetSuite) TestSnapSetFailOnStrictJSONAndString(c *check.C) {
+	_, err := snapset.Parser(snapset.Client()).ParseArgs([]string{"set", "snapname", "-t", "-s", "key={}"})
+	c.Assert(err, check.ErrorMatches, "cannot use -t and -s together")
+}
+
+func (s *snapSetSuite) TestSnapSetStrictString(c *check.C) {
```

Comment:
(nitpick) This test is not about `Strict` but `String` so maybe `TestSnapSetAsString`? Or am I misreading this?

Reply:
I think meant it as "strictly as a string" meaning not to be parsed but it's indeed confusing. I'll rename the two tests

Reply:
Aha, that makes sense but yeah, maybe still worth tweaking the name. Anyway, really tiny nitpick in any case :)